src/es/hackstore/src/eu/kanade/tachiyomi/animeextension/es/hackstore/Hackstore.kt

```diff
@@ -195,22 +197,35 @@ class Hackstore : ConfigurableAnimeSource, ParsedAnimeHttpSource() {
     override fun episodeFromElement(element: Element): SEpisode { throw UnsupportedOperationException() }
 
     // ============================ Video Links =============================
+    private fun extractUrlFromDonFunction(fullUrl: String): String {
+        val client = OkHttpClient()
+
+        val request = Request.Builder().url(fullUrl).build()
+        val response = client.newCall(request).execute()
+        val body = response.body.string()
+
+        val document = Jsoup.parse(body)
+
+        val scriptElement = document.select("script:containsData(function don())").firstOrNull()
+        val urlPattern = Regex("window\\.location\\.href\\s*=\\s*'([^']+)'")
+        val matchResult = scriptElement?.data()?.let { urlPattern.find(it) }
+        return matchResult?.groupValues?.get(1) ?: "url not found"
+    }
+
     override fun videoListParse(response: Response): List<Video> {
         val document = response.asJsoup()
         val videoList = mutableListOf<Video>()
 
-        val table = document.select("table.episodes")
-        val rows = table.select("tbody tr.tabletr")
-
-        rows.forEach { row ->
-            val serverElement = row.select("td:eq(0)")
-            val linkElement = row.select("td:eq(1) a")
+        val tabs = document.select("ul.TbVideoNv li.pres")
 
-            val server = serverElement.text()
-            val url = linkElement.attr("href")
-
-            val isLatino = server.contains("latino")
-            val isSub = server.contains("subtitulado")
+        tabs.forEach { tab ->
+            val server = tab.select("a.playr").text()
+            val deco = tab.select("a.playr").attr("data-href")
+            val langs = tab.select("a.playr").attr("data-lang")
+            val fullUrl = baseUrl + deco
```

Comment:
```suggestion
            val fullUrl = tab.select("a.playr").attr("abs:data-href")
            val langs = tab.select("a.playr").attr("data-lang")
```

Reply:
I tried this change, but it doesn't get the url